plugin/core/sessions.py

```diff
@@ -802,6 +803,16 @@ def run_command(self, command: Mapping[str, Any]) -> Promise:
             )
         )
 
+    def _apply_workspace_edit_async(self, edit: Any) -> Promise:
+        """
+        Apply workspace edits, and return a promise that resolves on the async thread again after the edits have been
+        applied.
+        """
+        changes = parse_workspace_edit(edit)
+        return Promise.on_main_thread() \
+            .then(lambda _: apply_workspace_edit(self.window, changes)) \
+            .then(Promise.on_async_thread)
```

Comment:
You don't have to do anything about it for now but we should improve Promise types so that we can indicate what value type it returns.

In this case, it will pass the value from `apply_workspace_edit` which will be an array with `None` but in this case we don't care about the return value so it should be just `None`.